### PR TITLE
Set default option as false when override existed folder using init command

### PR DESCRIPTION
**Pull request by qinezh**
#544 Set default option as false and notice user with Yellow literal when override current folder in init command

@superyyrrzz @hellosnow @ansyral @vwxyzh @chenkennt

### CODE REVIEW

File: src/docfx/SubCommands/InitCommand.cs [hunk 1]
```diff
@@ -406,6 +417,25 @@ private static bool Converter(string input)
             }
         }
 
+        /// <summary>
+        /// the default option is No
+        /// </summary>
+        private sealed class NoOrYesQuestion : SingleChoiceQuestion<bool>
+        {
+            private const string NoAnswer = "No";
```
Comment (hellosnow) [inline src/docfx/SubCommands/InitCommand.cs]: >NoAnswer [](start = 33, length = 8)

can it share with YesOrNoQuestion?

File: src/docfx/SubCommands/InitCommand.cs [hunk 2]
```diff
@@ -384,18 +384,29 @@ private static bool ProcessOverrideQuestion(string message)
                         overrides = false;
                     }
                 });
-            overrideQuestion.Process(null, new QuestionContext());
+
+            if (overrides)
+            {
+                overrideQuestion.Process(null, new QuestionContext {NeedWarning = true});
```
Comment (hellosnow) [inline src/docfx/SubCommands/InitCommand.cs]: >overrideQuestion.Process(null, new QuestionContext {NeedWarning = true}); [](start = 16, length = 73)

how about remove if/else and write as overrideQuestion.Process(null, new QuestionContext {NeedWarning = overrides});

File: src/docfx/SubCommands/InitCommand.cs [hunk 3]
```diff
@@ -584,6 +614,7 @@ private sealed class QuestionContext
         {
             public bool Quite { get; set; }
```
Comment (hellosnow) [inline src/docfx/SubCommands/InitCommand.cs]: >Quite [](start = 24, length = 5)

Should be quiet...

File: src/docfx/SubCommands/InitCommand.cs [hunk 4]
```diff
@@ -406,6 +417,25 @@ private static bool Converter(string input)
...
         }
 
+        /// <summary>
+        /// the default option is No
+        /// </summary>
+        private sealed class NoOrYesQuestion : SingleChoiceQuestion<bool>
+        {
+            private const string NoAnswer = "No";
+            private const string YesAnswer = "Yes";
+            private static readonly string[] YesOrNoAnswer = { NoAnswer, YesAnswer };
```
Comment (hellosnow) [inline src/docfx/SubCommands/InitCommand.cs]: >YesOrNoAnswer [](start = 45, length = 13)

NoOrYesAnswer?

File: src/docfx/SubCommands/InitCommand.cs [hunk 5]
```diff
@@ -384,18 +384,29 @@ private static bool ProcessOverrideQuestion(string message)
...
 
+        /// <summary>
+        /// the default option is Yes
+        /// </summary>
         private sealed class YesOrNoQuestion : SingleChoiceQuestion<bool>
         {
             private const string YesAnswer = "Yes";
             private const string NoAnswer = "No";
-            private static readonly string[] YesOrNoAnswer = new string[] { YesAnswer, NoAnswer };
+            private static readonly string[] YesOrNoAnswer = { YesAnswer, NoAnswer };
```
Comment (hellosnow) [inline src/docfx/SubCommands/InitCommand.cs]: >YesOrNoAnswer [](start = 45, length = 13)

how about extract the class and expost the string array as abstract property?

Comment (qinezh) [inline src/docfx/SubCommands/InitCommand.cs]: No need in this case?

File: src/docfx/SubCommands/InitCommand.cs [hunk 6]
```diff
@@ -384,18 +384,28 @@ private static bool ProcessOverrideQuestion(string message)
...
+
+        /// <summary>
+        /// the default option is Yes
+        /// </summary>
         private sealed class YesOrNoQuestion : SingleChoiceQuestion<bool>
         {
-            private const string YesAnswer = "Yes";
-            private const string NoAnswer = "No";
-            private static readonly string[] YesOrNoAnswer = new string[] { YesAnswer, NoAnswer };
+            private const string YesAnswer = YesOrNoOption.YesAnswer;
```
Comment (hellosnow) [inline src/docfx/SubCommands/InitCommand.cs]: >YesAnswer [](start = 33, length = 9)

remove the const here and reuse YesOrNoOption?

File: src/docfx/SubCommands/InitCommand.cs [hunk 7]
```diff
@@ -384,18 +384,28 @@ private static bool ProcessOverrideQuestion(string message)
                         overrides = false;
                     }
                 });
-            overrideQuestion.Process(null, new QuestionContext());
+
+            overrideQuestion.Process(null, new QuestionContext {NeedWarning = overrides});
```
Comment (hellosnow) [inline src/docfx/SubCommands/InitCommand.cs]: >{NeedWarning = overrides} [](start = 63, length = 25)

better indent like { NeedWarning = overrides }?

**Comment by chenkennt** [comment]
> - name: Articles

Remove the first blank line? It's a bit weird to see a blank line at the beginning of the file.

---
Refers to: src/docfx/SubCommands/InitCommand.cs:301 in c4dcb29. [](commit_id = c4dcb2907500cfed938d30fc078845b4e65969aa, deletion_comment = False)

File: src/docfx/SubCommands/InitCommand.cs [hunk 8]
```diff
@@ -375,7 +375,7 @@ private static void SaveConfigFile(string path, object config)
         private static bool ProcessOverrideQuestion(string message)
```
Comment (chenkennt) [inline src/docfx/SubCommands/InitCommand.cs]: >Override [](start = 35, length = 8)

should be overwrite?

**Comment by chenkennt** [comment]
:shipit:

**Comment by hellosnow** [comment]
:shipit:

**Pull request merged**